sgx-types/src/ssa.rs

```diff
@@ -330,14 +331,29 @@ impl Footer {
 ///
 /// Section 38.9, Table 38-7
 #[derive(Debug)]
-#[repr(C)]
+#[repr(C, align(4096))]
 pub struct StateSaveArea {
     ///  Section 38.9, Table 38-7
     pub xsave: XSave,
     /// Section 38.9, Table 38-7
     pub footer: Footer,
 }
 
+impl StateSaveArea {
+    /// Returns the size of the SSA in 4k pages, rather than bytes. According to
+    /// the documentation, SSAFrameSize (used in SECS) is referenced in pages.
+    /// If the SSAFrameSize is not 4k page aligned, returns None. See section
+    /// 38.7 for page-size requirement for SECS and Table 38-7 for requirement
+    /// that the SSA is page aligned.
+    pub fn frame_size() -> Option<NonZeroU32> {
```

Comment:
Don't use option here. We're defining a method on the `StateSaveArea` type. We'll just make sure it has the right alignment. I'd also implement the body as:

```rust
assert!(size_of::<StateSaveArea>() >= align_of::<StateSaveArea>());
let pages = size_of::<StateSaveArea>() / align_of::<StateSaveArea>();
unsafe { NonZeroU32::new_unchecked(pages) }
```

Then you can make the function `const`.

Reply:
It looks like we have to choose between the `const` and the assertion:
```
error[E0723]: loops and conditional expressions are not stable in const fn
   --> src/ssa.rs:349:9
       |
349 |         assert!(size_of::<StateSaveArea>() >= align_of::<StateSaveArea>());
       |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

https://github.com/rust-lang/rust/issues/51999

Reply:
@lkatalin Ditch the assertion. We control the size of the `StateSaveArea`. It is our problem.